src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidationHandler.java

```diff
@@ -11,35 +13,63 @@
 import uk.ac.ebi.subs.validator.data.structures.SingleValidationResultStatus;
 import uk.ac.ebi.subs.validator.data.structures.ValidationAuthor;
 
+import java.io.File;
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
 @Service
+@Getter
 @RequiredArgsConstructor
 public class FileContentValidationHandler {
 
     @NonNull
-    private FileContentValidator fileContentValidator;
+    private FastqFileValidator fastqFileValidator;
+
+    @NonNull
+    private VcfFileValidator vcfFileValidator;
 
     @NonNull
     private CommandLineParams commandLineParams;
 
+    @NonNull
+    private CommandLineParamChecker commandLineParamChecker;
+
     @NonNull
     private RabbitMessagingTemplate rabbitMessagingTemplate;
 
     private static final String EVENT_VALIDATION_SUCCESS = "validation.success";
     private static final String EVENT_VALIDATION_ERROR = "validation.error";
 
     public void handleFileContentValidation() throws IOException, InterruptedException {
-        List<SingleValidationResult> singleValidationResultList = fileContentValidator.validateFileContent();
+        List<SingleValidationResult> singleValidationResultList = commandLineParamChecker.validateParameters(commandLineParams);
+
+        if (singleValidationResultList.isEmpty()){
+            singleValidationResultList = doValidation();
+        }
+
 
         SingleValidationResultsEnvelope singleValidationResultsEnvelope =
                 generateSingleValidationResultsEnvelope(singleValidationResultList);
 
         sendValidationMessageToAggregator(singleValidationResultsEnvelope);
     }
 
+    private List<SingleValidationResult> doValidation() throws IOException, InterruptedException {
+        FileType fileType = commandLineParams.getFileTypeEnum();
+
+        if (fileType == FileType.FASTQ){
+            return fastqFileValidator.validateFileContent();
+        }
+        if (fileType == FileType.VCF){
+            return vcfFileValidator.validateFileContent();
+        }
+
+        throw new IllegalArgumentException("we have not implemented a handler for "+fileType);
```

Comment:
Should not this be a custom NotSupportedTypeException exception and we would throw it when we are not supporting a file type? We could support a file type and not having a content validator. In that case we just return a Passed validation result. What about this idea?

Reply:
I agree, switched to passing any types that we haven't implemented validation for.